packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -278,20 +372,22 @@ export class Table extends Construct {
         return this.keySchema.find(prop => prop.keyType === keyType);
     }
 
-    private addKey(name: string, type: AttributeType, keyType: string) {
+    private addKey(attribute: Attribute, keyType: string) {
         const existingProp = this.findKey(keyType);
         if (existingProp) {
-            throw new Error(`Unable to set ${name} as a ${keyType} key, because ${existingProp.attributeName} is a ${keyType} key`);
+            throw new Error(`Unable to set ${attribute.name} as a ${keyType} key, because ${existingProp.attributeName} is a ${keyType} key`);
         }
-        this.registerAttribute(name, type);
+        this.registerAttribute(attribute);
         this.keySchema.push({
-            attributeName: name,
+            attributeName: attribute.name,
             keyType
         });
         return this;
     }
 
-    private registerAttribute(name: string, type: AttributeType) {
+    private registerAttribute(attribute: Attribute) {
```

Comment:
Can we rename this to “validateAttribute” or something like that?

Also add documentation explaining what this method does

Reply:
I think "register" comes from the following snippet, and `this.attributeDefinitions` is used as an input parameter for table resource. Thus, IMHO, the original naming looks reasonable. 

```
if (!existingDef) {
    this.attributeDefinitions.push({
        attributeName: name,
        attributeType: type
    });
 }
```

It is a good idea to have documentation here. I will add it.